wikichange/src/content/content.js

```diff
@@ -94,79 +102,147 @@ let innerHTML = wikiText.innerHTML;
-            innerHTML.substring(index, index + text.length) +
-            "</mark>" +
-            innerHTML.substring(index + text.length);
-        wikiText.innerHTML = innerHTML;
-    }
+  let index = innerHTML.indexOf(text);
+  if (index >= 0) {
+    innerHTML =
+      innerHTML.substring(0, index) +
+      `<mark style='background-color: ${color}'>` +
+      innerHTML.substring(index, index + text.length) +
+      "</mark>" +
+      innerHTML.substring(index + text.length);
+    wikiText.innerHTML = innerHTML;
+  }
 };
 
 /* Highlights the words that are given with context. No support for links yet */
 const highlightContentWithContext = (json, color) => {
-    let foundIndex = -1;
-    let controlIndex = 0;
-    let highlight = json["highlight"];
-    let before = json["content_before"];
-    let after = json["content_after"];
-    while ((foundIndex = innerHTML.indexOf(highlight, controlIndex)) != -1) {
-        let afterText = innerHTML.substring(foundIndex + highlight.length, foundIndex + highlight.length + after.length);
-        let beforeText = innerHTML.substring(foundIndex - before.length, foundIndex);
-        console.log("After " + afterText);
-        console.log("Before " + beforeText);
-        if (afterText === after && beforeText === before) {
-            innerHTML =
-                innerHTML.substring(0, foundIndex) +
-                `<mark style='background-color: ${color}'>` +
-                innerHTML.substring(foundIndex, foundIndex + highlight.length) +
-                "</mark>" +
-                innerHTML.substring(foundIndex + highlight.length);
-            wikiText.innerHTML = innerHTML;
-            break;
+  let foundIndex = -1;
+  let controlIndex = 0;
+  let highlight = json["highlight"];
+  let before = json["content_before"];
+  let after = json["content_after"];
+  while ((foundIndex = innerHTML.indexOf(highlight, controlIndex)) != -1) {
+    let afterText = innerHTML.substring(
+      foundIndex + highlight.length,
+      foundIndex + highlight.length + after.length
+    );
+    let beforeText = innerHTML.substring(
+      foundIndex - before.length,
+      foundIndex
+    );
+    console.log("After " + afterText);
+    console.log("Before " + beforeText);
+    if (afterText === after && beforeText === before) {
+      innerHTML =
+        innerHTML.substring(0, foundIndex) +
+        `<mark style='background-color: ${color}'>` +
+        innerHTML.substring(foundIndex, foundIndex + highlight.length) +
+        "</mark>" +
+        innerHTML.substring(foundIndex + highlight.length);
+      wikiText.innerHTML = innerHTML;
+      break;
+    }
+    controlIndex = foundIndex + highlight.length;
+  }
+};
+
+/* Highlights the words that are given with context. Support for links, 
+there are some edge cases that don't work yet (highlight is link + no link) or 
+context and highlight are links */
+const highlightContentUsingNodes = (context, color) => {
```

Comment:
So from my understanding, this is an improved version of the previous highlight where it also highlights content in form of links?

Reply:
I would want to see a more obvious example of what can this version archive that the previous one (I assume it is `highlightContentWithContext`) couldn't. I might also be out of touch since it has been so long since I thought about this lol.